### ISSUE TITLE
Thank you!

### ISSUE THREAD

**Issue by asigalov61** [PatricioGuinle/CoffeMIDI#1]
Hey Patricio,

Thank you very much for creating CoffeeMIDI/proper MIDI Search. This was long overdue and I really hope your project will be noticed.

I wanted to ask you if you plan to implement a custom MIDI search option? It would be cool if you could upload your own MIDI to classify and match.

I also wanted to ask you if you would like to colab? It would be cool to do the entire LAKH dataset or similar so that classification is more precise and so that there is more music to listen to.

Let me know.

Sincerely,

Alex

**Comment by PatricioGuinle** [comment]
Hi Alex!

Has been a great surprise to find your message.

CoffeMIDI has been the first job I did while studying data science

The original version, not implemented in web, has the option of loading a MIDI to be compared with the complete dataset. It would be nice to implement it in a future version.

Also, You are right about the LAKH dataset
I have taken hundreds of MIDI files from different places, only filtering for similarity I found a good number of duplicates and I have filtered them and finally I am left with 90k. I think that with it I could also solve the lack of a really descriptive name. Can you tell me more about the dataset and your experience with it ?

I also have seen a little of your work and I found very interesting. For sure it would be good to collaborate. Tell me more about what you have in mind.

Greetings

Patricio

**Comment by asigalov61** [comment]
Hey Patricio,

You are very welcome :) You did a good job with Coffe MIDI :)

Yes, LAKH dataset is very nice. It has all annotations and everything. It is also de-duped and contains most pub MIDIs. Its about 170k MIDIs. 

I already built a very simple and efficient classifier based on LAKH. Check it out:
https://github.com/asigalov61/tegridy-tools/blob/main/tegridy-tools/notebooks/Instant_Classy_MIDI_Classifier.ipynb

It uses a simple match by MIDI notation (start-time, duration, channel, pitch, velocity). This way it is possible to achieve 70% accuracy with minimum compute costs for such a large dataset. Its not as advanced as your matching system, but it is very fast and simple.

RE: Collabing: I mostly wanted to see if I can help you with Coffe MIDI? I can't do web stuff, but I can do Python stuff. And I can also help with hosting I think. I was thinking Pythonanywhere would be a good hosting place for a project like that.

Let me know.

Alex

**Comment by PatricioGuinle** [comment]
I'm checking Instant_Classy_MIDI_Classifier, thanks  !

have you already seen the website?
it is quite simple, visually precarious, but you can see the model API working.
Check finding different gender results.
I am open to returns and points for improvement.

http://coffemidi.com.ar/

Because of how I have modularized the work, implementing a new dataset does not cause me a lot of work, only processing time. we can exchange our contacts in order to talk it better. I am also open to thinking about new projects.

**Comment by asigalov61** [comment]
Yes, I really liked the website and the interface :) Very nice :)

I would only suggest adding a custom MIDI option (IMHO it's very important for a system like that). And also I would clean up the names/descriptions of the files because some of them have paths in them and also extensions. In my classifier, I only leave the base file name of the MIDI. This way, it is clear what song it is.

Yes, I know that processing can be difficult, especially on a large dataset. I can help with that if you want.

And sure, if you have anything else in mind, let me know. I am primarily interested in Music AI/Algorithmic Music, so anything related would be great. :)

My email is asigalov61@hotmail.com if you want to contact me directly.

Alex.

**Comment by PatricioGuinle** [comment]
I think your AI knowledge and talent deserve at least one gmail hahah

great! I'm going to write you there

**Comment by asigalov61** [comment]
LOL 🙂

I hate Google tbh 🙁They are not a very ethical company no more 🙁


________________________________
